### PR TITLE
docs: make docs scale well on all screen sizes + responsive menu

### PR DESCRIPTION
**Pull request by htor**
update: this pr also fixes https://github.com/supercollider/supercollider/issues/3940

the docs are not scaled when displayed on a tablet or mobile phone, so it is hard to read and use.
this change fixes that and makes the home/logo link and the doc title collapse on smaller screens.
i didn't want to change the way the menu works (i.e. by making a mobile-version or something) because people are used to it and i know there has been recent work done on it.

on mobile now:
screen shot 2018-07-18 at 01 16 19

on mobile after:
screen shot 2018-07-18 at 01 16 34

the menu re-appears fully when the viewport grows:
_users_htor_projects_contrib_supercollider_helpsource_browse html 2

### CODE REVIEW

**Comment by htor** [comment]
@snappizz you're welcome i forgot to add the renderer - it's now added. are there more renderers/html-outputting files i need to know of?

another thing, 
the front page has "Table of Contents" in the menu. i didn't calculate that one in this pull, so the menu wraps on that page (see screenshot under). could it be a good idea to merge "Indexes" and "Table of Contents" into "Menu", to get more space on small screens? also, i think TOC should be on all pages, not just the front page.

**Comment by prko** [comment]
Hi, I am reworking #3865 now.
Should I add the new lines in scdoc.css by htor integrate? Or it will be done automatically?

**Comment by htor** [comment]
@prko hi, this branch is not ready yet (need some discussion) so don't pull it into yours please. i don't know the preferred way, but it seems the sc team likes to pick commits and do all the merges themselves?

**Comment by prko** [comment]
ok. thanks1

**Comment by snappizz** [comment]
thanks @htor. this is looking good, one last step is to heed the advice of the turtle at the top of `SCDocRenderer.sc` and increment `SCDoc.version`.

> are there more renderers/html-outputting files i need to know of?

nope, i think you got all of them. it's definitely quite annoying that we have to go and edit half a dozen files when we want to make html changes -- it would be cool if the SCDoc renderer also generated all the html files that we're manually editing.

**Comment by htor** [comment]
@snappizz ah, i didn't actually listen to the turtle. will increment version on next commit.
about that TOC-question, what do you think? is it possible to merge?

**Comment by brianlheim** [comment]
@snappizz - what do you think about merging TOC & "Indexes" into "Menu" on smaller screens? I think this is reasonable as long as it isn't too hard to do.

**Comment by prko** [comment]
What do you think about using an abbreviation or a  single glyph for each item in the menubar?
- browser: 
  - abbreviation
    - Br, Brs. Brws?
  - a glyph: 📃 or 📜 or U+1F5D0
    - https://unicode-table.com/en/1F5D0/
    - https://unicode-table.com/en/1F4C3/
    - https://unicode-table.com/en/1F5D0/
- search: 
  - abbreviatieon
    - Sr, Srch?
    - https://www.allacronyms.com/search/abbreviated
  - a glyph: 🔍 
    - http://www.fileformat.info/info/unicode/char/1f50d/index.htm
- indexes
  - abbreviatieon
    - Id, Idx?
    - https://www.allacronyms.com/index/abbreviated
  - a glyph: 🔎 or  📇 or 🗂
    - https://unicode-table.com/en/search/?q=index
    - https://unicode-table.com/en/2315/
    - http://www.fileformat.info/info/unicode/char/1f50e/index.htm

**Comment by brianlheim** [comment]
@prko I would support the shortening of Table of Contents to TOC. The rest are not commonly encountered and likely to be confusing.

**Comment by prko** [comment]
> I would support the shortening of Table of Contents to TOC

Thank you for supporting this.

I think "SuperCollider Version Number" is not necessary because there is the SC-cube on the left side of it.
untitled 6


A long title of a document makes the menubar in two lines. It is sometimes confusing.
screenshot 2018-08-09 10 27 20

The shortening of Table of Contents to TOC and the removal of "SuperCollider Version Number" will help to avoid this.

**Comment by snappizz** [comment]
no strong preference here either -- if it saves space, feel free to shorten to TOC and/or remove version number.

File: HelpSource/scdoc.css [hunk 1]
```diff
@@ -1,8 +1,13 @@
+*, *::before, *::after {
+    box-sizing: inherit;
+}
+
 html, body {
+    position: relative;
+    box-sizing: border-box;
```
Comment (htor) [inline HelpSource/scdoc.css]: border-box makes browsers include padding in the total computed with / height of elements. so when we say `#foo { height: 33px; }` it's guaranteed to be 33px high regardless of padding. very convenient and commonly used in most modern css.

Comment (prko) [reply HelpSource/scdoc.css]: Thanks!

**Comment by htor** [comment]
could someone test this pr and tell me if it works good or not?

**Comment by prko** [comment]
I would like to test it, but I do not know how I test your PR.
I could manually reflect what you updated, but there is a conflict SCClassLibrary/SCDoc/SCDoc.sc.

**Comment by htor** [comment]
@prko 
to test other pr, just make sure you have the git remote `upstream` set up (see CONTRIBUTING.md) 
then you basically do:
```
git pull upstream pull/3891/head:topic/docs-responsive
git checkout docs-responsive
```
now you can build and test the pr on the `docs-responsive` branch locally.

**Comment by prko** [comment]
Sorry for the late response.
I did it in El Capitan as follows:

Firstly I had done:
```
pko-MBP3:supercollider prko$ git clone --recursive https://github.com/prko/supercollider.git;

git remote add upstream https://github.com/supercollider/supercollider.git

pko-MBP3:supercollider prko$ cd supercollider

pko-MBP3:supercollider prko$ git remote -v

pko-MBP3:supercollider prko$ git pull upstream pull/3891/head:topic/docs-responsive

pko-MBP3:supercollider prko$ git checkout topic/docs-responsive

```
then I did:
```
pko-MBP3:supercollider prko$ mkdir -p build

pko-MBP3:supercollider prko$ cd build

pko-MBP3:build prko$ cmake -G Xcode -DCMAKE_PREFIX_PATH=`brew --prefix qt5` -DSUPERNOVA=ON ..

pko-MBP3:build prko$ cmake --build . --target install --config RelWithDebInfo

```
The followings are the screenshots in Chrome.
screenshot 2018-09-19 14 37 54

I think it works!

**Comment by prko** [comment]
@htor 
There is still a conflicting file. I think it blocks the next step.

**Comment by htor** [comment]
> @htor
> There is still a conflicting file. I think it blocks the next step.

thanks, fixed!

**Comment by htor** [comment]
@snappizz 
howdy ho! sure this can't make it into 3.10?

**Comment by snappizz** [comment]
i think it can! i'll try to look at it in the next few days, ping me if i forget.

**Pull request merged**